### DISCUSSION TITLE
Caching mechanism implementation

### DISCUSSION THREAD

**Discussion by pokyah** [koenderks/rcityviews#26] [Ideas]
Hi @koenderks ,

Thanks for this excellent package ! 

I have implemented a caching mechanism using a wrapper function inside `cityview` function based on `memoise` package. This allows for faster rebuild of the map while playing around with different themes and also alleviates the OSM requested servers. 

If you agree, can I submit a PR ? 

Thanks 

PS = I have also added themes inspired from python prettymaps package

**Comment by koenderks** [comment]
Seems interesting! You can make a PR if you want. I'm curious about how you implemented this, hopefully I can look at it sometime this week(end).

**Reply by pokyah** [reply]
Thanks ! Will do it right now :)
